src/main/java/org/graylog/integrations/notifications/types/SlackEventNotification.java

```diff
@@ -38,145 +38,143 @@
 import java.util.List;
 import java.util.Locale;
 import java.util.Map;
+import java.util.stream.Collectors;
 
 import static com.google.common.base.Strings.isNullOrEmpty;
 import static java.util.Objects.requireNonNull;
 
 public class SlackEventNotification implements EventNotification {
 
-	public interface Factory extends EventNotification.Factory {
-		@Override
+    private static final Logger LOG = LoggerFactory.getLogger(SlackEventNotification.class);
+    private final EventNotificationService notificationCallbackService;
+    private final Engine templateEngine;
+    private final NotificationService notificationService;
+    private final ObjectMapper objectMapper;
+    private final NodeId nodeId;
+    private final SlackClient slackClient;
+
+    @Inject
+    public SlackEventNotification(EventNotificationService notificationCallbackService,
+                                  ObjectMapper objectMapper,
+                                  Engine templateEngine,
+                                  NotificationService notificationService,
+                                  NodeId nodeId, SlackClient slackClient) {
+        this.notificationCallbackService = notificationCallbackService;
+        this.objectMapper = requireNonNull(objectMapper);
+        this.templateEngine = requireNonNull(templateEngine);
+        this.notificationService = requireNonNull(notificationService);
+        this.nodeId = requireNonNull(nodeId);
+        this.slackClient = requireNonNull(slackClient);
+    }
+
+    /**
+     * @param ctx
+     * @throws EventNotificationException is thrown when sending is failed
+     */
+    @Override
+    public void execute(EventNotificationContext ctx) throws EventNotificationException {
+        final SlackEventNotificationConfig config = (SlackEventNotificationConfig) ctx.notificationConfig();
+        LOG.debug("SlackEventNotification backlog size in method execute is [{}]", config.backlogSize());
+
+        try {
+            SlackMessage slackMessage = createSlackMessage(ctx, config);
+            slackClient.send(slackMessage, config.webhookUrl());
+        } catch (Exception e) {
+
+            LOG.error("SlackEventNotification send error for id {} : {}", ctx.notificationId(), e.toString());
+            final Notification systemNotification = notificationService.buildNow()
+                    .addNode(nodeId.toString())
+                    .addType(Notification.Type.GENERIC)
+                    .addSeverity(Notification.Severity.NORMAL)
+                    .addDetail("SlackEventNotification send error ", e.toString());
+
+            notificationService.publishIfFirst(systemNotification);
+            throw new EventNotificationException("Slack notification is triggered, but sending failed. " + e.getMessage(), e);
```

Comment:
This exception handling block is catching `Exception`. Since the `SlackClient#send` method is already returning a `TemporaryEventNotificationException` or `PermanentEventNotificationException`, we need a separate exception handling block that catches `EventNotificationException`s.

Otherwise we wrap the `EventNotificationException` returned from `SlackClient#send` in another `EventNotificationException` here and the scheduler will never retry a `TemporaryEventNotificationException`. That means notifications can be lost.